File: tests/03_ibj_1.go
package main

import . "glyph/tests/interp"

func t() {
    c := []uint64{
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0xffffffffffffffff,
    }
    i := []uint16{
        CPU_encode_op_ibj(64),
        CPU_encode_op_movib_i64(0,0),
        CPU_encode_op_break(0),
    }
    CPU_test("ibl_1", c, i);
}

func main() {
    t()
}

File: tests/03_ibj_1.c
#include "glyph.h"

void t()
{
    u64 c[] = {
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0xffffffffffffffffull,
    };
    u16 i[] = {
        cpu_encode_op_ibj(64),
        cpu_encode_op_movib_i64(0,0),
        cpu_encode_op_break(0),
    };
    cpu_test("ibl_1", c, i);
}

int main(int argc, char **argv)
{
    t();
    return 0;
}
